nomos-libp2p/src/behaviour/chainsync/swarm_ext.rs

```diff
@@ -0,0 +1,49 @@
+use std::collections::HashSet;
+
+use cryptarchia_sync::{ChainSyncError, HeaderId, SerialisedBlock, SerialisedHeaderId};
+use futures::stream::BoxStream;
+use libp2p::PeerId;
+use tokio::sync::{mpsc::Sender, oneshot};
+
+use crate::{behaviour::BehaviourError, Swarm};
+
+impl Swarm {
+    pub fn request_tip(
+        &self,
+        peer_id: PeerId,
+        reply_sender: oneshot::Sender<Result<SerialisedHeaderId, ChainSyncError>>,
+    ) -> Result<(), BehaviourError> {
+        let Some(chain_sync) = self.swarm.behaviour().chain_sync.as_ref() else {
+            return Err(BehaviourError::OperationNotSupported);
+        };
+
+        chain_sync
+            .request_tip(peer_id, reply_sender)
+            .map_err(Into::into)
+    }
+
+    pub fn start_blocks_download(
+        &self,
+        peer_id: PeerId,
+        target_block: HeaderId,
+        local_tip: HeaderId,
+        latest_immutable_block: HeaderId,
+        additional_blocks: HashSet<HeaderId>,
+        reply_sender: Sender<BoxStream<'static, Result<SerialisedBlock, ChainSyncError>>>,
```

Comment:
I think it would be nice to have at least a type alias for this type as it is a bit bit:


```suggestion
        reply_sender: Sender<SerializedBlockStream>,
```